### ISSUE TITLE
Utility for testing the Google API keys

### ISSUE THREAD

**Issue by lindseydiloreto** [doublesecretagency/craft-googlemaps#8]
Within the context of the control panel, it would be nice to have a page where people can **diagnose issues with their Google API keys**.

 - Are the keys valid?
 - Do they have access to the right APIs?
 - Are they restricted to server or browser?

... and any other questions which may need to be answered.

We need a page where someone can easily see if their Google Maps keys are properly configured. And if they aren't configured properly, it should be obvious as to what needs to be done about it.

---

> Migrated from Smart Map repo: https://github.com/doublesecretagency/craft-smartmap/issues/38

**Comment by AndrewMac** [comment]
Hi Lindsey, any action on this by any chance? I happened to see this suggestion when I came to look to see if anyone else was having issues. For some reason my site is no longer showing the map. Or rather, the map appears then disappears and is replaced with this message:

```
Sorry! Something went wrong.
This page didn't load Google Maps correctly. See the JavaScript console for technical details
```

Funny thing is, it used to work, and is still working on the local dev site. API key details seem correct on Google Console, and in any case they haven't changed since it was working ... which was just a few weeks ago. I think it might have stopped working after a recent Google Maps minor update, but I can't be completely sure.

Anyway, thanks in advance for any help or advice you may be able to give with this.

Cheers, Andrew

**Comment by lindseydiloreto** [comment]
Hmm... I don't have any insights, but I do agree that a debug page would be helpful! 😄 

It does sound like an API key issue. Feel free to DM me on Discord, and we can debug it the old-fashioned way.

**Comment by lindseydiloreto** [comment]
Good news, the "Test Google API Keys" utility has been released in **v4.3.2**! 🎉 

Here's the relevant [documentation...](https://plugins.doublesecretagency.com/google-maps/guides/test-google-api-keys/)
